CenterForOpenScience/exp-addons
# Set permissions for new session collection when appropriate

Refs https://openscience.atlassian.net/browse/LEI-115

## Purpose
When a session is created, allow loggedin-as-jam users to create records in that collection.

## Notes for reviewers
You will need to update your Jam server before running this. Otherwise, when creating a new experiment, you will get an error that permissions must be an integer.

## Test cases
SETUP:
- Create a new experiment (using the experimenter "new experiment" button on the list page) and populate it with sample structure for the player to run (copy an existing experiment's structure into the new one, using the edit experiment page)
  - Make one experiment with the permissions sent to the server, and one without (...suppress the new feature by commenting out those lines in the code)
  - Make sure the experiment is active

- Login as an OSF user, and confirm that at the end of playing through a newly created experiment via the preview route, you can submit session data. (because your permissions are inherited from the namespace)
   - [x] With the new branch
   - [x] Without the new branch

- Login as a Jam user and load a newly created experiment.
  - [x] Without the session permissions code, you should NOT be able to create a session for this experiment (you will be allowed to perform the experiment, but get a 403 forbidden error from the session collection when saving)
  - [x] With the session permissions code, you SHOULD be able to create and save a session for this experiment
